### ISSUE TITLE
Web interface not working

### ISSUE THREAD

**Issue by marco44** [gerbera/gerbera#425]
Hi,

I'm using the archlinux AUR package (https://aur.archlinux.org/packages/gerbera/).

When trying to run gerbera (after following all the steps indicated by the package, such as generating and editing the configuration file), the web interface doesn't work: there is a weird blue area appearing at the bottom of the screen, with an exclamation mark followed by a "{" character (I can try to do a screenshot if this helps).

As I'm no web developper, I have no idea what's going wrong. This is on an ARM-32 server, by the way. There is no error in the system journal...

It occurs with both firefox and chromium, and in safe mode too... Firefox is 
65, chromium 72.

All objects have the 200 status in the network log of firefox's debugger.

The only messages I get from the debugger's console are
```
Password fields present on an insecure (http://) page. This is a security risk 
that allows user login credentials to be stolen.[Learn More]
```
and
```
Synchronous XMLHttpRequest on the main thread is deprecated because of its 
detrimental effects to the end user’s experience. For more help http://
xhr.spec.whatwg.org/
```
Both are warnings...

As said before, I just don't know where to look for, not being at all a web developper :)

Regards

**Comment by elmodaddyb** [comment]
The 2 warnings in the console are expected. However the display you mention is not. 

Based on your description it sounds like the browser is not accurately hiding elements. This could be do to an invalid package or something amiss with the JavaScript. 

I will try to download the AUR package to see what happens. It will take some time to create a vm with the setup you mention.

**Comment by marco44** [comment]
Ok. The AUR works for other people though. So there must be something weird in my env...

**Comment by v00d00** [comment]
Which version of libupnp are you using? Can you post a debug log?

**Comment by marco44** [comment]
libupnp18 1.8.4. Here is the log...

[gerbera_debug.txt](https://github.com/gerbera/gerbera/files/2860436/gerbera_debug.txt)

**Comment by elmodaddyb** [comment]
I played around with the UI workflow a bit to try and produce the issue you mention.  There is a slight flash of the `toast` close icon when the page loads...but not what you describe.

> If I had to guess, it seems like something is wrong with the buffer memory sizes or memory references when sending the bytestream to the client.

As @v00d00 mentioned libupnp is the webserver serving up files.  I wonder if you have to install the latest git version from AUR [libupnp-git](https://aur.archlinux.org/packages/libupnp-git/)

**Comment by marco44** [comment]
Ok, got it. I had no idea libupnp was used to serve the web interface, so I didn't mention it.

As libupnp 1.8 is only available in AUR, I had to build it myself, and got this: ```#error libupnp uses large file support, so users must do that, too```. Searching for this error brought me here: https://github.com/gerbera/gerbera/issues/218 . And I passed the --disable-largefile suggested in the post. Which was a VERY bad idea.

libupnp 1.8 itself builds perfectly without this. What doesn't build is the sample program provided with it… because glibc still doesn't build with large file support by default (which I don't understand, given how long it's been here, and how long 2GB+ files have been around).

Anyway, for people having to build libupnp by themselves on 32 bits systems, the real solution is simply to pass ```CFLAGS+=' -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE '``` to the environment variables before starting to build libupnp (setting this on your PKGBUILD for instance).

I just rebuilt, and now the interface works perfectly (well, it's still indexing files, as the machine isn't very powerful :) )

I hope this will help other users who might face this. In the meantime, I'll try to convince the libupnp 1.8 AUR maintainer to add this... I don't know if there would be a way to detect this and have gerbera warn users, because it's quite tricky ?
